RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -97,18 +98,24 @@ public bool Generate(NuGetReference analyzeRef, string language, string sqaleFil
                 definition.PackageId = package.Id;
                 definition.PackageVersion = package.Version.ToString();
                 definition.Manifest = CreatePluginDefinition(package);
-                definition.RulesFilePath = Path.Combine(baseDirectory, "rules.xml");
 
                 string packageDir = this.packageHandler.GetLocalPackageRootDirectory(package);
 
                 definition.StaticResourceName = Path.GetFileName(packageDir) + ".zip";
                 definition.SourceZipFilePath = Path.Combine(baseDirectory, definition.StaticResourceName);
                 ZipFile.CreateFromDirectory(packageDir, definition.SourceZipFilePath, CompressionLevel.Optimal, false);
-                
-                bool success = TryGenerateRulesFile(this.packageHandler.LocalCacheRoot, definition.RulesFilePath, language, packageDir);
 
-                if (success)
+                IEnumerable<DiagnosticAnalyzer> analyzers = GetAnalyzers(packageDir, this.packageHandler.LocalCacheRoot, language);
+
+                if (analyzers.Any())
                 {
+                    definition.RulesFilePath = GenerateRulesFile(analyzers, baseDirectory);
+
+                    if (definition.SqaleFilePath == null)
+                    {
+                        definition.SqaleFilePath = GenerateFixedSqaleFile(analyzers, baseDirectory);
+                    }
+
```

Comment:
Both the rule generation and sqale generation require the list of analyzers to traverse. I've refactored the old "TryGenerateRulesFile" method to separate out the analyzer discovery into a separate method.